dashboard/test/ui/features/projects/personal_project_gallery.feature

```diff
@@ -10,7 +10,57 @@ Scenario: Can Toggle to the Public Project Gallery
   Given I am on "http://studio.code.org/projects"
   And I wait until element "#react-personal-projects" is visible
   And element "#public-gallery" is not visible
-  Then I click selector "#uitest-gallery-switcher div:contains(Public Projects)"
-  Then check that I am on "http://studio.code.org/projects/public"
-  And I wait until element "#public-gallery" is visible
-  And element "#react-personal-projects" is not visible
+  Then I navigate to the public gallery via the gallery switcher
+
+Scenario: Can Publish and Unpublish a Project (Button Version)
+  Given I make a playlab project named "Publishable Project"
+  Given I am on "http://studio.code.org/projects"
+  And I wait until element "#react-personal-projects" is visible
+  And I wait until element ".ui-personal-projects-table" is visible
+  And the project table contains 1 row
+  And the first project in the table is named "Publishable Project"
+  Then I publish the project from the personal projects table publish button
+  Then I click selector ".ui-personal-projects-unpublish-button"
+  And I wait until element ".ui-personal-projects-publish-button" is visible
+
+Scenario: Can Rename a Project
+  Given I make a playlab project named "Old Name"
+  Given I am on "http://studio.code.org/projects"
+  And I wait until element "#react-personal-projects" is visible
+  And I wait until element ".ui-personal-projects-table" is visible
+  And the project table contains 1 row
+  And the first project in the table is named "Old Name"
+  Then I click selector ".ui-projects-table-dropdown"
+  And I press the child number 0 of class ".pop-up-menu-item"
+  And I wait until element ".ui-project-rename-input" is visible
+  And I wait for 5 seconds
+  And I type "New Name" into ".ui-project-rename-input"
+  And I wait for 5 seconds
+  Then I click selector ".ui-projects-rename-save"
+  And I wait for 5 seconds
+  And the first project in the table is named "New Name"
```

Comment:
This step is failing and it's boggling why. It looks like the new name is typed in the input box, save is pressed and then the old name appears!?!? This is a gif of the test running: 
![new name](https://user-images.githubusercontent.com/12300669/44880916-da67f100-ac7b-11e8-9ea1-d76e1ec6845e.gif)

@epeach I remember that you waged a similar war once with keystrokes not registering in a UI test or something along those lines; any ideas here?

Reply:
That's maddening! I ended up using steps like "I press the first 'xyz' element" and "I press keys 'input' for element 'xyz'". Here: https://github.com/code-dot-org/code-dot-org/blob/8baef740ebddc680448e51465263328bb46bddb6/dashboard/test/ui/features/feedback_tab.feature#L35

That might help?

Reply:
If that doesn't work, it would be interesting to see which part of this is breaking down - like are key strokes not registering, is the update display not working, etc.?

Reply:
My suspicion is that because the text input is a "controlled component", it's not enough to change the text as viewed in the browser -- you also have to get react to notice the text, which it only does when a certain event (presumably the change event) is fired.

If ErinP's suggestion doesn't work, I would suggest using these steps instead of `I press keys ...` because they explicitly fire the change event:
https://github.com/code-dot-org/code-dot-org/blob/2d72ef033d2a8ac580de395ca3afe0d6796caf9d/dashboard/test/ui/features/step_definitions/steps.rb#L483-L497